### DISCUSSION TITLE
appDir: on-demand revalidate bug. Showing one step old stale data.

### DISCUSSION THREAD

**Discussion by surjithctly** [vercel/next.js#43967] [Help]
I'm using Sanity to update content. But each time it updates, it returns one step old data. Not fresh. See below video. 

### Video transcription:
Starts next production builds
change “first post” to “Our first post”
Simulate “webhook” requests
Data do not change
[next attempt]
change “Our first post” to “Next first post”
Data change to Stale “Our First post” this time
but new data is “Next First Post” (edited) 

https://user-images.githubusercontent.com/1884712/207079785-48b821f7-1da7-439b-a07c-665fcdbe3689.mp4

**Comment by wwwdepot** [comment]
Did you revalidate the "home" page?

```js
  await res.revalidate(`/`);
  await res.revalidate(`/${slug}`);
```

**Reply by surjithctly** [reply]
I revalidated both home and post pages. 

I guess this is because sanity apicdn updates after webhook fires.

**Comment by nowchris** [comment]
Hi there, I'm having this _exact_ issue. Did you ever solve this @surjithctly and, if so, do you remember how?

**Reply by surjithctly** [reply]
The reason was, I still had `apicdn: true` on production which caused the issue. make sure its false on production as well.
